src/transformers/generation_flax_utils.py

```diff
@@ -237,12 +241,23 @@ def generate(
             >>> print("Generated:", tokenizer.batch_decode(outputs, skip_special_tokens=True))
         """
         # set init values
-        max_length = max_length if max_length is not None else self.config.max_length
+        if max_length is not None and max_new_tokens is not None:
+            raise ValueError("Both `max_new_tokens` and `max_length` are defined which is incorrect, use only one.")
+        if max_length is None:
```

Comment:
Could we reduce this to two lines?

```suggestion
        prompt_length = 1 if self.config.is_encoder_decoder else input_ids.shape[1]
        max_length = prompt_length + max_new_tokens if max_new_tokens is not None else self.config.max_length
```
```

Reply:
IMO this is a bit more readable and would avoid a double nested if - else

Reply:
Fair enough, I tend to prefer simpler lines usually but in that particular case I think you are right that inlined if is more readable.